uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsenceType.php

```diff
@@ -540,4 +540,39 @@ public static function getOneWithMustTakePublicHolidayAsLeaveRequest() {
 
     return null;
   }
+
+  /**
+   * Returns the date the TOIL will expire based on the date given to the method.
+   *
+   * @param int $typeId
+   *   The Absence type ID
+   * @param string $date
+   *   The date to calculate TOIL expiry date
+   *
+   * @return null|false|string
+   */
+  public static function calculateToilExpiryDate($typeId, $date) {
+    $absenceType = new self();
+    $absenceType->id = $typeId;
+    $absenceType->find(true);
+
+    if (!$absenceType->allow_accruals_request) {
+      return null;
+    }
+    if (!$absenceType->accrual_expiration_unit || !$absenceType->accrual_expiration_duration){
```

Comment:
the advantage of the method is that it can be reused on other places and it hides the logic of how we know that toil never expires. Say we change things and add a toil_never_expire future, then, instead of fixing all the places were we use accrual_expiration_unit and accrual_expiration_duration, we would only need to change the method